prow/plugins/transfer-issue/transfer-issue.go

```diff
@@ -0,0 +1,96 @@
+)
+
+func init() {
+	plugins.RegisterGenericCommentHandler(pluginName, handleGenericComment, helpProvider)
+}
+
+func helpProvider(_ *plugins.Configuration, _ []config.OrgRepo) (*pluginhelp.PluginHelp, error) {
+	pluginHelp := &pluginhelp.PluginHelp{
+		Description: "The transfer-issue plugin transfers a GitHub issue from one repo to another in the same organization.",
+	}
+	pluginHelp.AddCommand(pluginhelp.Command{
+		Usage:       "/transfer-issue <destination repo in same org>",
+		Description: "Transfers an issue to a different repo in the same org.",
+		Featured:    true,
+		WhoCanUse:   "Org members.",
+		Examples:    []string{"/transfer-issue kubectl"},
+	})
+	return pluginHelp, nil
+}
+
+type githubClient interface {
+	GetRepo(org, name string) (github.FullRepo, error)
+	CreateComment(org, repo string, number int, comment string) error
+	IsCollaborator(org, repo, user string) (bool, error)
+	TransferIssue(org, dstRepoNodeID string, issueNodeID string) (*github.TransferIssueMutation, error)
+}
+
+func handleGenericComment(pc plugins.Agent, e github.GenericCommentEvent) error {
+	// TODO: check if this is an actual comment on an issue?
+	if e.Action != github.GenericCommentActionCreated {
+		return nil
+	}
+	matches := transferRe.FindAllStringSubmatch(e.Body, -1)
+	if len(matches) == 0 {
+		return nil
+	}
+	if len(matches) != 1 && len(matches[0]) != 1 {
+		// TODO: post this as a comment and log
+		return fmt.Errorf("/transfer-issue must only be used once and with a single destination repo")
+	}
+	return handleTransfer(pc.GitHubClient, pc.Logger, e, matches[0][0])
+}
+
+func handleTransfer(gc githubClient, log *logrus.Entry, e github.GenericCommentEvent, dstRepoName string) error {
+	org := e.Repo.Owner.Login
+	srcRepo := e.Repo.Name
+	user := e.User.Login
+
+	dstRepo, err := gc.GetRepo(org, dstRepoName)
+	if err != nil {
+		log.WithError(err).Errorf("could not fetch destination repo: %s/%s", org, dstRepo)
+		// TODO: post comment. error here means something went wrong or that the repo doesn't exist (404)
+	}
+
+	isCollaboratorSrc, err := gc.IsCollaborator(org, srcRepo, user)
+	if err != nil {
+		log.WithError(err).Errorf("could not fetch if user: %s is collaborator of source repo: %s/%s", user, org, srcRepo)
+		// TODO: comment something went wrong?
+	}
+	isCollaboratorDst, err := gc.IsCollaborator(org, dstRepoName, user)
+	if err != nil {
+		log.WithError(err).Errorf("could not fetch if user: %s is collaborator of destination repo: %s/%s", user, org, dstRepoName)
+		// TODO: comment something went wrong?
+	}
+	if !(isCollaboratorSrc && isCollaboratorDst) {
+		// TODO: comment user is not able to transfer because they aren't an org member
+	}
+
+	issue, err := gc.TransferIssue(org, dstRepo.NodeID, e.NodeID)
+	if err != nil {
+		log.WithError(err).Errorf("could not transfer issue: from: to:")
+		// TODO: comment that issue could not be transferred
+	}
+	// TODO: is commenting at this point necessary or a race condition?
```

Comment:
Personally I think it's not necessary, the comment itself serves as a record of who transferred the issue when to where. The only missing bit of information is *from* where it was transferred. 

Does GH itself show some kind of event about the transfer in the UI?

Reply:
Thanks for the review!

I think you're right. GitHub does do a pretty good job of surfacing this https://github.com/eddiezane/test-repo-2/issues/3

Reply:
It even preserves the journey if you ping pong it back and forth.